File: A4KPI/Models/AccountGroup.cs
using A4KPI.Models.Interface;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace A4KPI.Models
{
    [Table("AccountGroups")]
    public class AccountGroup: IDateTracking
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; }
        public int Position { get; set; }
        public int Sequence { get; set; }
        public int CreatedBy { get; set; }
        public int ModifiedBy { get; set; }
        public DateTime CreatedTime { get ; set ; }
        public DateTime? ModifiedTime { get ; set ; }
        public virtual ICollection<Period> Periods { get; set; }
    }
}

File: App/src/app/_core/_model/account.group.ts
import { Account } from "./account";
import { Period } from "./period";

export interface AccountGroup {
  id: number;
  name: string;
  createdBy: number;
  modifiedBy: number;
  createdTime: string;
  position: number;
  sequence: number;
  modifiedTime: string | null;
  accounts: Account[];
  periods: Period[];
}
